### DISCUSSION TITLE
how can I show uploadfiles and its urls use infolist builder?

### DISCUSSION THREAD

**Discussion by baituocxy** [filamentphp/filament#13200] [Help] [answered]
### Package

Infolist builder

### Package Version

v3.2.88

### How can we help you?

does their have any  entry  like FileEntry  in infolist builder to handle FileUpload?
I use TextEntry,  but its not convenient to show urls when upload file is multiple, thanks

**Comment by leandrocfe** [comment]
maybe [repeatable](https://filamentphp.com/docs/3.x/infolists/entries/repeatable) when it is multiple?

**Comment by baituocxy** [comment, marked as answer]
`
TextEntry::make('attach')->label('attach')->columnSpan(6)
                            ->placeholder('none')
                            ->listWithLineBreaks()
                            ->bulleted()
                            ->formatStateUsing(function ($state) {
                                return sprintf('<span style="--c-50:var(--primary-50);--c-400:var(--primary-400);--c-600:var(--primary-600);"  class="text-xs rounded-md mx-1 font-medium px-2 min-w-[theme(spacing.6)] py-1  bg-custom-50 text-custom-600 ring-custom-600/10 dark:bg-custom-400/10 dark:text-custom-400 dark:ring-custom-400/30">
        <a href="%s"  target="_blank">%s</a></span>', '/storage/'.$state, basename($state));
})
->html() 
`

**Comment by isach5** [comment]
```
Infolists\Components\TextEntry::make('attachment')
                            ->label('Attachment')
                            ->placeholder('None')
                            ->listWithLineBreaks()
                            ->formatStateUsing(fn($state) => $state ? '<a href="' . asset('storage/agency/' . $state) . '" target="_blank" class="text-blue-600 hover:underline">View Attachment ' . basename($state) . '</a>' : 'No attachment')
                            ->html()
                            ->badge(),
```
Make sure to edit the path
